### ISSUE TITLE
Clang issuing incorrect diagnostic under C23+pedantic

### ISSUE THREAD

**Issue by Th3T3chn0G1t** [llvm/llvm-project#89396]
`clang -std=gnu23 -pedantic` enables `-Wgnu-binary-literal` despite binary literals being standardised in C23. This seems incorrect.

**Expected Behaviour:**
Standardized binary literals are not marked as an extension by `-pedantic`

**Actual Behaviour:**
`-Wgnu-binary-literal` is set so `error: binary integer literals are a GNU extension` diagnostic is issued

**Version/Env:**
```
> clang -v
clang version 18.1.3
Target: x86_64-pc-windows-msvc
Thread model: posix
InstalledDir: C:\Program Files\LLVM\bin
```

**Comment by tbaederr** [comment]
This is fixed in the next clang version: https://github.com/llvm/llvm-project/issues/72017
